catalyst-gateway/bin/src/service/api/cardano/rbac/mod.rs

```diff
@@ -1,5 +1,56 @@
 //! RBAC endpoints.
 
-pub(crate) mod chain_root_get;
-pub(crate) mod registrations_get;
-pub(crate) mod role0_chain_root_get;
+use poem_openapi::{param::Query, payload::Json, OpenApi};
+
+use crate::service::common::{
+    auth::{none_or_api_key::NoneOrApiKey, none_or_rbac::NoneOrRBAC},
+    tags::ApiTags,
+    types::cardano::query::stake_or_voter::StakeOrVoter,
+};
+
+mod rbac_registration;
+mod registrations_get;
+mod unprocessable_content;
+
+/// Cardano RBAC API Endpoints
+pub(crate) struct Api;
+
+#[OpenApi(tag = "ApiTags::Cardano")]
+impl Api {
+    #[oai(
+        path = "/draft/rbac/registrations/:catalyst_id",
+        method = "get",
+        operation_id = "rbacRegistrations"
+    )]
+    /// Get RBAC registrations
+    ///
+    /// This endpoint returns RBAC registrations by provided auth Catalyst Id credentials
+    /// or by the `lookup` argument if provided.
+    async fn rbac_registrations_get(
+        &self,
+        /// Stake address to get the RBACE registration for.
+        Query(lookup): Query<Option<StakeOrVoter>>,
```

Comment:
```suggestion
        Query(lookup): Query<Option<CatIdOrStake>>,
```

Voter has nothing to do with RBAC.  You can either lookup the latest RBAC registration which is tied to a known stake address (CIP-19 formatted like `stake1u94ullc9nj9gawc0899...` or  a catalyst id like `id.catalyst://cardano/FftxFnOrj2qmTuB2oZG2v0YEWJfKvQ9Gg8AgNAhDsKE` or`cardano/FftxFnOrj2qmTuB2oZG2v0YEWJfKvQ9Gg8AgNAhDsKE`  we should accept either the full URI or short form for catalyst id.

If `lookup` is not supplied we use the id given by the token to the `auth` parameter.

*Yes the suggested `CatIdOrStake` type doesn't exist, it will need to be created.*